mod-source-record-manager-server/src/main/java/org/folio/services/util/QueryPathUtil.java

```diff
@@ -0,0 +1,27 @@
+package org.folio.services.util;
+
+import javax.ws.rs.BadRequestException;
+
+import org.folio.Record;
+
+public class QueryPathUtil {
+
+  private QueryPathUtil() {
+    throw new IllegalStateException("Utility class");
+  }
+
+  /**
+   * Convert Sttong param to Folio RecordType enum
+   *
+   * @param recordType String param from URL path
+   * @return org.folio.Record.RecordType
+   */
+  public static Record.RecordType convert(String recordType) {
+    switch (recordType) {
+      case "marc-bib":
+        return Record.RecordType.MARC_BIB;
+      case "marc-holdings": return Record.RecordType.MARC_HOLDING;
+      default: throw new BadRequestException("Only marc-bib or marc-holdings supports");
+    }
```

Comment:
Create a pre-populated map in your class that contains the mapping between string and record type. 
For exception will be better to throw it in your controller or just without exception check for nullable.
```suggestion
    return Optional.ofNullable(queryToRecordTypeMap.get(recordType));
```
Also, method should have better naming, for example `toRecordType(String queryParam)`